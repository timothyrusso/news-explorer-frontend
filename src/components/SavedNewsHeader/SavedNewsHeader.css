.saved-news-header {
  min-height: 374px;
  width: 100%;
  box-sizing: border-box;
  padding: 120px 0 56px 104px;
}

.saved-news-header__intro {
  margin: 0;
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 1.33;
  color: rgba(26, 27, 34, 0.5);
}

.saved-news-header__title {
  font-family: 'Roboto Slab';
  font-style: normal;
  font-weight: 400;
  font-size: 40px;
  line-height: 1.15;
  color: #1A1B22;
  max-width: 530px;
  margin: 30px 0 28px;
}

.saved-news-header__keywords {
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 1.33;
  color: #1A1B22;
  margin: 0;
}
