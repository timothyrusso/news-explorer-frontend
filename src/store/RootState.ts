import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
