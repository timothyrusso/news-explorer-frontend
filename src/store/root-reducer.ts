import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
