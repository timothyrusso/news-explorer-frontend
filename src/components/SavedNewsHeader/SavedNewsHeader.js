import './SavedNewsHeader.css';
import { useSelector } from 'react-redux';

const SavedNewsHeader = () => {
  const currentUser = useSelector((state) => state.user.currentUser);
  const savedArticles = useSelector((state) => state.article.savedArticles);
  const searchKeywordsList = useSelector(
    (state) => state.user.searchKeywordsList
  );

  return (
    <header className="saved-news-header">
      <p className="saved-news-header__intro">Saved articles</p>
      <h1 className="saved-news-header__title">
        {currentUser.name}, you have {savedArticles.length} saved articles
      </h1>
      {searchKeywordsList.length >= 1 && (
        <p className="saved-news-header__keywords">
          By keywords:&nbsp;
          <span className="saved-news-header__keywords-bold">
            {searchKeywordsList.length >= 1 && searchKeywordsList[0]}
            {searchKeywordsList.length >= 2 && `, ${searchKeywordsList[1]}`}
            {searchKeywordsList.length === 3 &&
              `, and ${searchKeywordsList[2]}`}
            {searchKeywordsList.length > 3 &&
              `, and ${searchKeywordsList.length - 2} other`}
          </span>
        </p>
      )}
    </header>
  );
};

export default SavedNewsHeader;
