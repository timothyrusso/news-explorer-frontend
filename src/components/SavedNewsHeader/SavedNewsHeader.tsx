import React from 'react';
import { useSelector } from 'react-redux';
import type { RootState } from '../../store/RootState';
import './SavedNewsHeader.css';

const SavedNewsHeader = () => {
  const currentUser = useSelector((state: RootState) => state.user.currentUser);
  const savedArticles = useSelector(
    (state: RootState) => state.article.savedArticles
  );
  const searchKeywordsList = useSelector(
    (state: RootState) => state.user.searchKeywordsList
  );

  return (
    <header className="saved-news-header">
      <p className="saved-news-header__intro">Saved articles</p>
      <h1 className="saved-news-header__title">
        {currentUser &&
          'name' in currentUser &&
          `${currentUser.name}, you have ${savedArticles.length} saved articles`}
      </h1>
      {searchKeywordsList.length >= 1 && (
        <p className="saved-news-header__keywords">
          By keywords:&nbsp;
          <span className="saved-news-header__keywords-bold">
            {searchKeywordsList.length >= 1 && searchKeywordsList[0]}
            {searchKeywordsList.length >= 2 && `, ${searchKeywordsList[1]}`}
            {searchKeywordsList.length === 3 &&
              `, and ${searchKeywordsList[2]}`}
            {searchKeywordsList.length > 3 &&
              `, and ${searchKeywordsList.length - 2} other`}
          </span>
        </p>
      )}
    </header>
  );
};

export default SavedNewsHeader;
