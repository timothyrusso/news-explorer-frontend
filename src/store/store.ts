import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
