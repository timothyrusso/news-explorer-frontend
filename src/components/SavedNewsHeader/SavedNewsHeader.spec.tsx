import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import SavedNewsHeader from './SavedNewsHeader';

describe('SavedNewsHeader component', () => {
  it('Should render the SavedNewsHeader component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SavedNewsHeader />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
