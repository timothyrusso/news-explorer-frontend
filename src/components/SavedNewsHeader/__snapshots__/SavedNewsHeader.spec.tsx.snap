// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedNewsHeader component Should render the SavedNewsHeader component 1`] = `
<DocumentFragment>
  <header
    class="saved-news-header"
  >
    <p
      class="saved-news-header__intro"
    >
      Saved articles
    </p>
    <h1
      class="saved-news-header__title"
    />
  </header>
</DocumentFragment>
`;
